app/controllers/authentication.js

```diff
@@ -1,229 +1,238 @@
+    }
+  }
+
+  static github(req, res) {
+    const userApiUrl = 'https://api.github.com/user';
+    const accessTokenUrl = 'https://github.com/login/oauth/access_token';
 
     /*
       Authorize the user using github by exchanging authorization code for access token.
     */
-    var authorization = function (callback) {
-      var params = {
+    const authorization = (callback) => {
+      const params = {
         code: req.body.code,
         client_id: req.body.clientId,
         client_secret: nconf.get('github').clientSecret,
         redirect_uri: req.body.redirectUri,
       };
 
-      request.get({ url: accessTokenUrl, qs: params },
-        function (err, response, accessToken) {
-          if (err) {
-            winston.info('Github auth: authorization error');
-            callback({ status: 500, msg: err.toString() });
-          } else {
-            accessToken = qs.parse(accessToken);
-            var headers = { 'User-Agent': 'Satellizer' };
+      request.get({ url: accessTokenUrl, qs: params }, (err, response, accessToken) => {
+        if (err) {
+          winston.info('Github auth: authorization error');
+          callback({ status: 500, msg: err.toString() });
+        } else {
+          accessToken = qs.parse(accessToken);
+          const headers = { 'User-Agent': 'Satellizer' };
 
-            callback(null, accessToken, headers);
-          }
-        });
+          callback(null, accessToken, headers);
+        }
+      });
     };
 
     /*
       Retrieve the user's github profile.
     */
-    var getProfile = function (accessToken, headers, callback) {
-      request.get({ url: userApiUrl, qs: accessToken, headers, json: true },
-        function (err, response, profile) {
-          if (err) {
-            winston.info('Github auth: getProfile error');
-            callback({ status: 500, msg: err.toString() });
-          } else if (!err && response.statusCode !== 200) {
-            winston.info('Github auth: bad profile response');
-            callback({ status: 409, msg: 'Did not get github profile.' });
-          } else if (!profile.email) {
-            winston.info('Github auth: no email error');
-            callback({ status: 409, msg: 'Did not get email address.' });
-          } else {
-            callback(null, accessToken, headers, profile);
-          }
-        });
+    const getProfile = (accessToken, headers, callback) => {
+      request.get({ url: userApiUrl, qs: accessToken, headers, json: true }, (err, response, profile) => {
+        if (err) {
+          winston.info('Github auth: getProfile error');
+          callback({ status: 500, msg: err.toString() });
+        } else if (!err && response.statusCode !== 200) {
+          winston.info('Github auth: bad profile response');
+          callback({ status: 409, msg: 'Did not get github profile.' });
+        } else if (!profile.email) {
+          winston.info('Github auth: no email error');
```

Comment:
warn level. This is by the way problematic: GitHub default configuration use email as private so its not visible here. We have to change email as optional field probably..